### PR TITLE
Add role and task test helpers

### PR DESCRIPTION
**Pull request by adamgering**
Add role and task test helpers

Change unit mark to library mark.
Have library tests conform to strict linting

### CODE REVIEW

File: tests/rbac/common/manager/rbac_manager_test.py [hunk 1]
```diff
@@ -16,38 +16,36 @@
...
-
-        self.assertIsInstance(self.rbac.role, RoleManager)
-        self.assertTrue(callable(self.rbac.role.address))
-        self.assertTrue(callable(self.rbac.role.make))
-        self.assertTrue(callable(self.rbac.role.make_addresses))
-        self.assertTrue(callable(self.rbac.role.make_payload))
-        self.assertTrue(callable(self.rbac.role.create))
-        self.assertTrue(callable(self.rbac.role.send))
-        self.assertTrue(callable(self.rbac.role.get))
+        self.assertIsInstance(rbac.user, UserManager)
```
Comment (nadiabahrami) [inline tests/rbac/common/manager/rbac_manager_test.py]: So this is a test that is testing python and how it works, more then the actual code.  It is not necessary.  This is more for your knowledge then a remove.

File: tests/rbac/common/role/confirm_admin_test.py [hunk 2]
```diff
@@ -13,152 +13,136 @@
...
 
-    @pytest.mark.unit
+    @pytest.mark.library
     def test_make_payload(self):
         """Test making the message payload"""
-        self.assertTrue(callable(self.role.admin.confirm.make_payload))
-        message, admin_key = self.test_make()
-        payload = self.role.admin.confirm.make_payload(
-            message=message, signer_keypair=admin_key
+        self.assertTrue(callable(rbac.role.admin.confirm.make_payload))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_admin_test.py]: This line is not needed.  As the test below it is testing the 'callability' . of that method

File: tests/rbac/common/role/confirm_admin_test.py [hunk 3]
```diff
@@ -13,152 +13,136 @@
...
-    @pytest.mark.unit
+    @pytest.mark.library
     def test_make_addresses(self):
         """Test making the message addresses"""
-        self.assertTrue(callable(self.role.admin.confirm.make_addresses))
-        message, admin_key = self.test_make()
-
-        inputs, outputs = self.role.admin.confirm.make_addresses(
-            message=message, signer_keypair=admin_key
+        self.assertTrue(callable(rbac.role.admin.confirm.make_addresses))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_admin_test.py]: assertion not needed

File: tests/rbac/common/role/confirm_admin_test.py [hunk 4]
```diff
@@ -13,152 +13,136 @@
...
-        self.assertTrue(callable(self.role.admin.confirm.make))
-        role, _, keypair = self.get_testunit_user_role()
-        user = self.get_testdata_user()
-        reason = self.get_testdata_reason()
-        message = self.role.admin.confirm.make(
-            proposal_id=uuid4().hex,
-            user_id=user.user_id,
-            role_id=role.role_id,
-            reason=reason,
+        self.assertTrue(callable(rbac.role.admin.confirm.make))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_admin_test.py]: Not needed

File: tests/rbac/common/role/confirm_admin_test.py [hunk 5]
```diff
@@ -13,152 +13,136 @@
...
         self.assertIn(proposal_address, outputs)
         self.assertEqual(len(outputs), 2)
-        return payload
 
     @pytest.mark.integration
     def test_create(self):
         """Test executing the message on the blockchain"""
-        self.assertTrue(callable(self.role.admin.confirm.create))
-        self.assertTrue(callable(self.get_testdata_role_admin_proposal))
+        self.assertTrue(callable(rbac.role.admin.confirm.create))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_admin_test.py]: Not needed

File: tests/rbac/common/role/confirm_member_test.py [hunk 6]
```diff
@@ -13,153 +13,144 @@
...
-        self.assertTrue(callable(self.role.member.confirm.make))
-        role, _, keypair = self.get_testunit_user_role()
-        user = self.get_testdata_user()
-        reason = self.get_testdata_reason()
-        message = self.role.member.confirm.make(
-            proposal_id=uuid4().hex,
-            user_id=user.user_id,
-            role_id=role.role_id,
-            reason=reason,
+        self.assertTrue(callable(rbac.role.member.confirm.make))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_member_test.py]: Not needed

File: tests/rbac/common/role/confirm_member_test.py [hunk 7]
```diff
@@ -13,153 +13,144 @@
...
-    @pytest.mark.unit
+    @pytest.mark.library
     def test_make_addresses(self):
         """Test making the message addresses"""
-        self.assertTrue(callable(self.role.member.confirm.make_addresses))
-        message, owner_key = self.test_make()
-
-        inputs, outputs = self.role.member.confirm.make_addresses(
-            message=message, signer_keypair=owner_key
+        self.assertTrue(callable(rbac.role.member.confirm.make_addresses))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_member_test.py]: Not needed

File: tests/rbac/common/role/confirm_member_test.py [hunk 8]
```diff
@@ -13,153 +13,144 @@
...
 
-    @pytest.mark.unit
+    @pytest.mark.library
     def test_make_payload(self):
         """Test making the message payload"""
-        self.assertTrue(callable(self.role.member.confirm.make_payload))
-        message, owner_key = self.test_make()
-        payload = self.role.member.confirm.make_payload(
-            message=message, signer_keypair=owner_key
+        self.assertTrue(callable(rbac.role.member.confirm.make_payload))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_member_test.py]: Not needed

File: tests/rbac/common/role/confirm_owner_test.py [hunk 9]
```diff
@@ -13,153 +13,144 @@
...
-        self.assertTrue(callable(self.role.owner.confirm.make))
-        role, _, keypair = self.get_testunit_user_role()
-        user = self.get_testdata_user()
-        reason = self.get_testdata_reason()
-        message = self.role.owner.confirm.make(
-            proposal_id=uuid4().hex,
-            user_id=user.user_id,
-            role_id=role.role_id,
-            reason=reason,
+        self.assertTrue(callable(rbac.role.owner.confirm.make))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_owner_test.py]: Not needed

File: tests/rbac/common/role/confirm_owner_test.py [hunk 10]
```diff
@@ -13,153 +13,144 @@
...
-    @pytest.mark.unit
+    @pytest.mark.library
     def test_make_addresses(self):
         """Test making the message addresses"""
-        self.assertTrue(callable(self.role.owner.confirm.make_addresses))
-        message, owner_key = self.test_make()
-
-        inputs, outputs = self.role.owner.confirm.make_addresses(
-            message=message, signer_keypair=owner_key
+        self.assertTrue(callable(rbac.role.owner.confirm.make_addresses))
```
Comment (nadiabahrami) [inline tests/rbac/common/role/confirm_owner_test.py]: Not needed

File: tests/rbac/common/assertions/common.py [hunk 11]
```diff
@@ -47,7 +49,7 @@ def assertHasProp(self, item, name):
         self.assertTrue(
             hasattr(item, name), "Expected {}.{} to exist".format(item, name)
         )
-        prop = getattr(item, name)
+        getattr(item, name)
```
[the remark this answers is not in the document]
Comment (adamgering) [reply tests/rbac/common/assertions/common.py]: it tests that it is able to get the named attribute (does not raise an exception)

**Comment by adamgering** [comment]
All issues resolved or replied to.

File: tests/addressing/user.py [hunk 12]
```diff
@@ -23,7 +23,7 @@
 LOGGER = logging.getLogger(__name__)
 
 
-@pytest.mark.unit
+@pytest.mark.library
```
Comment (kthblmfld) [inline tests/addressing/user.py]: Not a show-stopper, but I am not seeing how we will be adopting/applying this at this time. 

What I can see is using pytest decorators to filter out tests we would run in a build pipeline (setting aside slow running tests for downstream/e2e tests to improve pre-merge build time). We can employ the decoration once it solves an existing problem. Without a problem being solved comes a reduced value-added.

File: tests/rbac/common/__init__.py [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+from tests.rbac.common.assertions import TestAssertions
+from tests.rbac.common.user.user_helper import UserTestHelper
+from tests.rbac.common.role.role_helper import RoleTestHelper
+from tests.rbac.common.task.task_helper import TaskTestHelper
+from tests.rbac.common.proposal.proposal_helper import ProposalTestHelper
+
+
+class TestHelper(TestAssertions):
+    def __init__(self, *args, **kwargs):
```
Comment (nadiabahrami) [inline tests/rbac/common/__init__.py]: Doc String on what this testhelper does

File: tests/rbac/common/addresser/address_assertions.py [hunk 14]
```diff
@@ -56,3 +59,12 @@ def assertIsAddress(self, value):
                 value
             ),
         )
+
+    def assertIsIdentifier(self, value):
+        self.assertIsInstance(value, str)
```
Comment (nadiabahrami) [inline tests/rbac/common/addresser/address_assertions.py]: doc s

File: tests/rbac/common/assertions/__init__.py [hunk 15]
```diff
@@ -13,11 +13,14 @@
...
 # -----------------------------------------------------------------------------
 
-from tests.rbac.common.sawtooth.batch_assertions import BatchAssertions
-from tests.rbac.common.user.user_helper import UserTestHelper
+from tests.rbac.common.assertions.batch import BatchAssertions
 
 
-class TestHelper(BatchAssertions):
+class TestAssertions(BatchAssertions):
     def __init__(self, *args, **kwargs):
```
Comment (nadiabahrami) [inline tests/rbac/common/assertions/__init__.py]: doc strings

File: tests/rbac/common/role/propose_admin_helper.py [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+# pylint: disable=invalid-name
+helper = TestHelper()
+
+
+class ProposeRoleAdminTestHelper(TestAssertions):
+    def __init__(self, *args, **kwargs):
+        TestAssertions.__init__(self, *args, **kwargs)
+
+    def id(self):
```
Comment (nadiabahrami) [inline tests/rbac/common/role/propose_admin_helper.py]: docstrings

File: tests/rbac/common/role/propose_member_helper.py [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+
+# pylint: disable=invalid-name
+helper = TestHelper()
+
+
+class ProposeRoleMemberTestHelper(TestAssertions):
+    def __init__(self, *args, **kwargs):
+        TestAssertions.__init__(self, *args, **kwargs)
+
+    def id(self):
```
Comment (nadiabahrami) [inline tests/rbac/common/role/propose_member_helper.py]: docstrings

File: tests/rbac/common/role/propose_owner_helper.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+# pylint: disable=invalid-name
+helper = TestHelper()
+
+
+class ProposeRoleOwnerTestHelper(TestAssertions):
+    def __init__(self, *args, **kwargs):
+        TestAssertions.__init__(self, *args, **kwargs)
+
+    def id(self):
```
Comment (nadiabahrami) [inline tests/rbac/common/role/propose_owner_helper.py]: docstrings

File: tests/rbac/common/role/propose_task_helper.py [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+
+# pylint: disable=invalid-name
+helper = TestHelper()
+
+
+class ProposeRoleTaskTestHelper(TestAssertions):
+    def __init__(self, *args, **kwargs):
+        TestAssertions.__init__(self, *args, **kwargs)
+
+    def id(self):
```
Comment (nadiabahrami) [inline tests/rbac/common/role/propose_task_helper.py]: docstring

File: tests/rbac/common/role/role_helper.py [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+class RoleTestHelper:
+    def __init__(self):
+        self.create_role = CreateRoleTestHelper()
+        self.admin = RoleAdminTestHelper()
+        self.owner = RoleOwnerTestHelper()
+        self.member = RoleMemberTestHelper()
+        self.task = RoleTaskTestHelper()
+
+    def id(self):
+        return self.create_role.id()
```
Comment (nadiabahrami) [inline tests/rbac/common/role/role_helper.py]: docs

File: tests/rbac/common/task/task_helper.py [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+        self.propose = ProposeTaskOwnerTestHelper()
+
+
+class TaskTestHelper:
+    def __init__(self):
+        self.create_task = CreateTaskTestHelper()
+        self.admin = TaskAdminTestHelper()
+        self.owner = TaskOwnerTestHelper()
+
+    def id(self):
```
Comment (nadiabahrami) [inline tests/rbac/common/task/task_helper.py]: docstrings

**Pull request merged**